src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
Right -- I'm not suggesting passing the sync.once variable but rather a function that calls it (again ```pipeData(clientConn, connOut, req, func() {closeOnce.Do(closeConns)})```), so that would preserve the sync.once intention. 

Your changes look OK, except that we need to call the closing function if either copy method returns. I believe your change will only reach the deferred func in intercept when the client->proxy copy ends and signal <- true is called.